lib/mini_twitter_client.rb

```diff
@@ -0,0 +1,63 @@
+require 'faraday'
+require 'json'
+
```

Comment:
So far we were using primitive types (hash, array, etc.) to represent data.

It has its downsides, because when I'm working with the Tweet that is represented as a hash, I can easily make a mistake and build hash with invalid format, for example:

```ruby
{
  author: {
    name: 'Jon',
    email: 'jon@local'
  },
  id: 1,
  mesage: "Hey!"
}
```

Notice that in above case, I made a typo in `message` ;)))

---

You can create more "advanced" type to represent Tweet:

```ruby
class Tweet
  def initialize(id: id, message: message, ...)
    # ...
  end

  def id
    @id
  end

  # ...
end
```

Then you could use it this way:

```ruby
tweet = Tweet.new(id: 1, message: 'Hey!', ...)
puts tweet.id
puts tweet.message
```

instead of:

```ruby
tweet = {id: 1, message: 'Hey!',...}
puts tweet[:id]
puts tweet[:message]
```

Look what would happen if you would try to get the value of unexpected attribute in both cases:

```ruby
# instance of Hash
tweet[ddfds]

# instance of Tweet class
tweet.ddfds
```